<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.42">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>3&nbsp; Verify Downloaded Data – CAMDA2025 metadatos</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./4_inpufile_MLandDL.html" rel="next">
<link href="./2_Preprocessing_CAMDApublicdata.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-2f5df379a58b258e96c21c0638c20c03.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-b53751a350365c71b6c909e95f209ed1.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-854d42f42d0f6b455b86ad16a464e311.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark-61e1ae932dc4076e99360d80d2eb001f.min.css" rel="prefetch" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet">
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet">
<script src="site_libs/datatables-binding-0.33/datatables.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet">
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet">
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet">
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./3_Download_SRAdata.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Verify Downloaded Data</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">CAMDA2025 metadatos</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/EveliaCoss/CAMDA2025_metadatos" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">CAMDA 2025</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Datos</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2_Preprocessing_CAMDApublicdata.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Comparing CAMDA metadata related with public data</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./3_Download_SRAdata.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Verify Downloaded Data</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./4_inpufile_MLandDL.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Generating the Input File for Machine Learning and Deep Learning Models</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#to-do-list" id="toc-to-do-list" class="nav-link active" data-scroll-target="#to-do-list"><span class="header-section-number">3.1</span> To do list</a></li>
  <li><a href="#acomodo-de-las-carpetas-por-especie" id="toc-acomodo-de-las-carpetas-por-especie" class="nav-link" data-scroll-target="#acomodo-de-las-carpetas-por-especie"><span class="header-section-number">3.2</span> Acomodo de las carpetas por especie</a>
  <ul>
  <li><a href="#test-information" id="toc-test-information" class="nav-link" data-scroll-target="#test-information"><span class="header-section-number">3.2.1</span> Test information</a></li>
  <li><a href="#training-information" id="toc-training-information" class="nav-link" data-scroll-target="#training-information"><span class="header-section-number">3.2.2</span> Training information</a></li>
  <li><a href="#sra-de-las-especies" id="toc-sra-de-las-especies" class="nav-link" data-scroll-target="#sra-de-las-especies"><span class="header-section-number">3.2.3</span> SRA de las especies</a></li>
  </ul></li>
  <li><a href="#eliminacion-de-secuencias-de-mala-calidad-y-de-adaptadores" id="toc-eliminacion-de-secuencias-de-mala-calidad-y-de-adaptadores" class="nav-link" data-scroll-target="#eliminacion-de-secuencias-de-mala-calidad-y-de-adaptadores"><span class="header-section-number">3.3</span> Eliminacion de secuencias de mala calidad y de adaptadores</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Verify Downloaded Data</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<blockquote class="blockquote">
<p>Aim: To confirm successful data download, folder structure organization, and assess the quality of the retrieved files.</p>
<ul>
<li><strong>Author</strong>: Evelia Coss</li>
<li><strong>Date</strong>: 1/July/2025</li>
<li><strong>Input files:</strong>
<ul>
<li><a href="https://github.com/EveliaCoss/CAMDA2025_metadatos/blob/main/quality_rawdata/SRA_names_carpetas.txt"><code>SRA_names_carpetas.txt</code></a>: Generated from all raw data downloaded to the cluster, located under <code>/files/CAMDA2025_AntibioticResistance/genome_reassembly/reads</code>. This file is stored in the <code>quality_metadata</code> directory.</li>
<li><a href="https://github.com/EveliaCoss/CAMDA2025_metadatos/blob/main/rawdata/TrainAndTest_cleaned/complete_metadata.RData"><code>complete_metadata.RData</code></a>: This RData file contains a consolidated set of key objects used in workflows for antimicrobial resistance analysis and sequencing metadata. It supports smooth transitions between preprocessing, integration, and analysis stages. See complete details in <a href="https://eveliacoss.github.io/CAMDA2025_metadatos/2_Preprocessing_CAMDApublicdata.html"><strong>Chapter 2</strong></a>.</li>
</ul></li>
<li><strong>Output files:</strong>
<ul>
<li><a href="https://github.com/EveliaCoss/CAMDA2025_metadatos/tree/main/quality_rawdata/species_ids_carpetas"><code>species_ids_carpetas folder</code></a>: This directory contains the <code>.txt</code> files generated by the <code>save_ids()</code> function. Each file corresponds to a single species and includes only the SRA accession IDs, separated by training and test sets. These files are used to organize and manage species-specific accessions for downstream processing.</li>
</ul></li>
<li><strong>Functions:</strong>
<ul>
<li><a href="https://github.com/EveliaCoss/CAMDA2025_metadatos/blob/main/functions/create_species_function.R"><code>create_species()</code></a>: Parses metadata to create species-specific variables containing accession IDs, species names, and an <code>info</code> status label:
<ul>
<li><code>Training_data_download, Training_data_missing, Test_data_download, Test_data_missing.</code></li>
</ul></li>
<li><a href="https://github.com/EveliaCoss/CAMDA2025_metadatos/blob/main/functions/Save_ids_files.R"><code>save_ids()</code></a>: Generates <code>.txt</code> files per species with only accession IDs, separated by training and test groups, based on the variables created by <code>create_species_function()</code>.</li>
</ul></li>
</ul>
</blockquote>
<p>Crear archivo solo de los datos que se encuentren descargados</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> raw/ <span class="at">-mindepth</span> 1 <span class="at">-maxdepth</span> 1 <span class="at">-type</span> d <span class="at">-exec</span> basename {} <span class="dt">\;</span> <span class="op">&gt;</span> SRA_names_carpetas.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Lo descargue en mi computadora dentro de la carpeta donde se respalda Github.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">scp</span> ecoss@elisa.matmor.unam.mx:/files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/SRA_names_carpetas.txt .</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<section id="to-do-list" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="to-do-list"><span class="header-section-number">3.1</span> To do list</h2>
<ul>
<li>Checar cuanto porcentage de descarga tenemos de los datos</li>
<li>Mover a sus respectivas carpetas por especies</li>
<li>Orden de las carpetas, clasificar por Herramienta/equipo de secuenciacion (Illumina, PacBio, Nanopore), Tamaños de los reads, Tipos de bibliotecas (PE/SE)</li>
<li>Mover los FastQC a sus respectivas carpeta y correr MultiQC</li>
</ul>
<p>Load packages</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT)        <span class="co"># Tablas bonitas</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Import Data</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load metadata cleaned and antibiogram information</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">here</span>(<span class="st">"rawdata/TrainAndTest_cleaned"</span>, <span class="st">"complete_metadata.RData"</span>))</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Load information about download rawdata, import as data.frame</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>SRA_down <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">here</span>(<span class="st">"quality_rawdata"</span>, <span class="st">"SRA_names_carpetas.txt"</span>), <span class="at">header =</span> <span class="cn">FALSE</span>, <span class="at">col.names =</span> <span class="st">"accession"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Clasificacion de los datos descargados</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Combina los datos de test y training en un solo data frame con una columna fuente</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>info_combined <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  test_completeInfo_db <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">info =</span> <span class="st">"Test_data"</span>),</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  training_cleanedInfo_db <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">info =</span> <span class="st">"Training_data"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(accession, scientific_name_CAMDA, info)</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Une con SRA_down y asigna 'missing' si no hay coincidencias</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>SRA_down_joined_data <span class="ot">&lt;-</span> SRA_down <span class="sc">%&gt;%</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(info_combined, <span class="at">by =</span> <span class="st">"accession"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">info =</span> <span class="fu">if_else</span>(accession <span class="sc">%in%</span> SRA_down<span class="sc">$</span>accession, <span class="fu">paste0</span>(info,<span class="st">"_download"</span>) , <span class="fu">paste0</span>(info,<span class="st">"_missing"</span>)))</span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(SRA_down_joined_data<span class="sc">$</span>info)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
           NA_download     Test_data_download      Test_data_missing 
                  2150                   3216                   2242 
Training_data_download  Training_data_missing 
                  3212                   2238 </code></pre>
</div>
</div>
<p>De que especies ya tenemos informacion</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>SRA_down_joined_data <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(scientific_name_CAMDA, info) <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_accession =</span> <span class="fu">n_distinct</span>(accession)) <span class="sc">%&gt;%</span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_accession))  <span class="sc">%&gt;%</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  DT<span class="sc">::</span><span class="fu">datatable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'scientific_name_CAMDA'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-f6ecbaade0d0f3335f31" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f6ecbaade0d0f3335f31">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37"],[null,"Neisseria gonorrhoeae","Neisseria gonorrhoeae","Klebsiella pneumoniae","Klebsiella pneumoniae","Salmonella enterica","Salmonella enterica","Staphylococcus aureus","Staphylococcus aureus","Pseudomonas aeruginosa","Pseudomonas aeruginosa","Campylobacter jejuni","Campylobacter jejuni","Escherichia coli","Escherichia coli","Streptococcus pneumoniae","Streptococcus pneumoniae","Acinetobacter baumannii","Acinetobacter baumannii","Streptococcus pneumoniae","Streptococcus pneumoniae","Acinetobacter baumannii","Acinetobacter baumannii","Klebsiella pneumoniae","Klebsiella pneumoniae","Escherichia coli","Escherichia coli","Salmonella enterica","Salmonella enterica","Pseudomonas aeruginosa","Pseudomonas aeruginosa","Staphylococcus aureus","Staphylococcus aureus","Neisseria gonorrhoeae","Neisseria gonorrhoeae","Campylobacter jejuni","Campylobacter jejuni"],["NA_download","Test_data_download","Training_data_download","Test_data_missing","Training_data_missing","Test_data_missing","Training_data_missing","Test_data_download","Training_data_download","Test_data_missing","Training_data_missing","Test_data_download","Training_data_download","Test_data_download","Training_data_download","Test_data_download","Training_data_download","Test_data_download","Training_data_download","Test_data_missing","Training_data_missing","Test_data_missing","Training_data_missing","Test_data_download","Training_data_download","Test_data_missing","Training_data_missing","Test_data_download","Training_data_download","Test_data_download","Training_data_download","Test_data_missing","Training_data_missing","Test_data_missing","Training_data_missing","Test_data_missing","Training_data_missing"],[2150,726,726,637,634,603,603,555,554,553,552,533,533,475,473,402,402,316,316,205,205,172,172,154,153,37,37,37,37,18,18,16,16,15,15,4,4]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>scientific_name_CAMDA<\/th>\n      <th>info<\/th>\n      <th>n_accession<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"scientific_name_CAMDA","targets":1},{"name":"info","targets":2},{"name":"n_accession","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="acomodo-de-las-carpetas-por-especie" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="acomodo-de-las-carpetas-por-especie"><span class="header-section-number">3.2</span> Acomodo de las carpetas por especie</h2>
<p>Esperaria contar con 9 carpetas, unas por cada especie, ademas de tener subcarpetas relacionadas con el equipo de secuenciacion (Illumina, Nanopore, PacBio), tipo de bibliotecas (paired-end, single-end), asi como de los tamaños de los reads. Viendose algo como eso:</p>
<pre><code>/raw_data/
├── Especie_1/
│   ├── illumina/
│   │   ├── paired_end/
│   │   └── single_end/
│   ├── pacbio/
│   │   └── single_end/
│   └── nanopore/
│       └── single_end/
├── Especie_2/
│   ├── illumina/
│   │   ├── paired_end/
│   │   └── single_end/
│   ├── pacbio/
│   │   └── single_end/
│   └── nanopore/
│       └── single_end/
...
└── Especie_9/
    ├── illumina/
    │   ├── paired_end/
    │   └── single_end/
    ├── pacbio/
    │   └── single_end/
    └── nanopore/
        └── single_end/</code></pre>
<p>Información sobre los tipos de bibliotecas:</p>
<ul>
<li><strong><code>PAIRED</code></strong>: indica que es una <strong>secuenciación por pares (paired-end)</strong>, es decir, se han secuenciado ambos extremos de un fragmento de ADN.</li>
<li><strong><code>SINGLE</code></strong>: significa que las lecturas son <strong>de un solo extremo</strong> (single-end), es decir, se secuencia solo un extremo de cada fragmento de ADN.</li>
</ul>
<p>Información sobrelas tecnologias de secuenciacion:</p>
<ul>
<li><strong><code>LS454</code></strong>: se refiere a la tecnología de secuenciación <strong>Roche/454 Life Sciences</strong>, una de las primeras plataformas de secuenciación masiva (ya descontinuada).</li>
<li><strong><code>PACBIO_SMRT</code></strong>: se refiere a la tecnología de <strong>Pacific Biosciences (PacBio)</strong> usando su plataforma <strong>SMRT (Single Molecule Real-Time)</strong> sequencing.</li>
<li><strong><code>ILLUMINA</code></strong>: Tecnología de secuenciación de segunda generación, que es la plataforma más común y extendida para NGS. Produce lecturas cortas (100-300 pb típicamente) con alta precisión y gran volumen.</li>
<li><strong><code>OXFORD_NANOPORE</code></strong>: Plataforma de secuenciación desarrollada por Oxford Nanopore Technologies, que produce lecturas muy largas en tiempo real mediante la detección de cambios en corriente eléctrica cuando el ADN pasa por un poro nanopore.</li>
</ul>
<section id="test-information" class="level3" data-number="3.2.1">
<h3 data-number="3.2.1" class="anchored" data-anchor-id="test-information"><span class="header-section-number">3.2.1</span> Test information</h3>
<p>Vamos a revisar que tipos de datos esperamos tener de las secuenciaciones para generar las carpetas relacionadas</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 1. Expandir accession IDs en antibiograms_db</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>antibiograms_expanded <span class="ot">&lt;-</span> antibiograms_db <span class="sc">%&gt;%</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate_rows</span>(accession, read_type, <span class="at">sep =</span> <span class="st">" "</span>)  <span class="co"># cambia el separador si es coma por row</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Unir con test_completeInfo_db por accession</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>test_joined_data <span class="ot">&lt;-</span> test_completeInfo_db <span class="sc">%&gt;%</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(antibiograms_expanded <span class="sc">%&gt;%</span> <span class="fu">select</span>(accession, read_type),</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="st">"accession"</span>)</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. Resultado</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(test_joined_data<span class="sc">$</span>read_type)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
       PAIRED_ILLUMINA           PAIRED_LS454     PAIRED_PACBIO_SMRT 
                 70121                    228                    457 
       SINGLE_ILLUMINA           SINGLE_LS454 SINGLE_OXFORD_NANOPORE 
                    27                    329                     14 
    SINGLE_PACBIO_SMRT 
                  1935 </code></pre>
</div>
</div>
<p>Distribucion por especies y tecnologias de secuenciacion</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>test_joined_data <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(scientific_name_CAMDA, read_type) <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_accession =</span> <span class="fu">n_distinct</span>(accession)) <span class="sc">%&gt;%</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_accession))  <span class="sc">%&gt;%</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  DT<span class="sc">::</span><span class="fu">datatable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'scientific_name_CAMDA'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-19f179bc9299fbc177bd" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-19f179bc9299fbc177bd">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32"],["Klebsiella pneumoniae","Neisseria gonorrhoeae","Salmonella enterica","Streptococcus pneumoniae","Pseudomonas aeruginosa","Staphylococcus aureus","Campylobacter jejuni","Escherichia coli","Acinetobacter baumannii","Campylobacter jejuni","Escherichia coli","Streptococcus pneumoniae","Pseudomonas aeruginosa","Staphylococcus aureus","Acinetobacter baumannii","Staphylococcus aureus","Klebsiella pneumoniae","Streptococcus pneumoniae","Streptococcus pneumoniae","Neisseria gonorrhoeae","Pseudomonas aeruginosa","Acinetobacter baumannii","Klebsiella pneumoniae","Klebsiella pneumoniae","Klebsiella pneumoniae","Escherichia coli","Salmonella enterica","Acinetobacter baumannii","Acinetobacter baumannii","Escherichia coli","Neisseria gonorrhoeae","Staphylococcus aureus"],["PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA",null,"SINGLE_PACBIO_SMRT",null,"SINGLE_PACBIO_SMRT","SINGLE_PACBIO_SMRT",null,null,null,"SINGLE_LS454","PAIRED_LS454","SINGLE_ILLUMINA","PAIRED_PACBIO_SMRT","SINGLE_PACBIO_SMRT","SINGLE_PACBIO_SMRT","PAIRED_PACBIO_SMRT","SINGLE_LS454",null,"SINGLE_PACBIO_SMRT","PAIRED_PACBIO_SMRT","SINGLE_OXFORD_NANOPORE","PAIRED_PACBIO_SMRT","SINGLE_PACBIO_SMRT","SINGLE_LS454"],[757,731,636,550,538,533,478,474,458,59,32,27,24,19,18,18,16,16,14,9,9,7,7,6,5,4,4,3,2,2,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>scientific_name_CAMDA<\/th>\n      <th>read_type<\/th>\n      <th>n_accession<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"scientific_name_CAMDA","targets":1},{"name":"read_type","targets":2},{"name":"n_accession","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="training-information" class="level3" data-number="3.2.2">
<h3 data-number="3.2.2" class="anchored" data-anchor-id="training-information"><span class="header-section-number">3.2.2</span> Training information</h3>
<p>Vamos a revisar que tipos de datos esperamos tener de las secuenciaciones para generar las carpetas relacionadas</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># 2. Unir con training_cleanedInfo_db por accession</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>training_joined_data <span class="ot">&lt;-</span> training_cleanedInfo_db <span class="sc">%&gt;%</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(antibiograms_expanded <span class="sc">%&gt;%</span> <span class="fu">select</span>(accession, read_type),</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">by =</span> <span class="st">"accession"</span>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 3. Resultado</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(training_joined_data<span class="sc">$</span>read_type)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code>
       PAIRED_ILLUMINA           PAIRED_LS454     PAIRED_PACBIO_SMRT 
                 70018                    228                    457 
       SINGLE_ILLUMINA           SINGLE_LS454 SINGLE_OXFORD_NANOPORE 
                    27                    329                     14 
    SINGLE_PACBIO_SMRT 
                  1920 </code></pre>
</div>
</div>
<p>Distribucion por especies y tecnologias de secuenciacion</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>training_joined_data <span class="sc">%&gt;%</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(scientific_name_CAMDA, read_type) <span class="sc">%&gt;%</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_accession =</span> <span class="fu">n_distinct</span>(accession)) <span class="sc">%&gt;%</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(n_accession))  <span class="sc">%&gt;%</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>  DT<span class="sc">::</span><span class="fu">datatable</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>`summarise()` has grouped output by 'scientific_name_CAMDA'. You can override
using the `.groups` argument.</code></pre>
</div>
<div class="cell-output-display">
<div class="datatables html-widget html-fill-item" id="htmlwidget-afec5f210ec702ea9fa3" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-afec5f210ec702ea9fa3">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32"],["Klebsiella pneumoniae","Neisseria gonorrhoeae","Salmonella enterica","Streptococcus pneumoniae","Pseudomonas aeruginosa","Staphylococcus aureus","Campylobacter jejuni","Escherichia coli","Acinetobacter baumannii","Campylobacter jejuni","Escherichia coli","Streptococcus pneumoniae","Pseudomonas aeruginosa","Staphylococcus aureus","Acinetobacter baumannii","Staphylococcus aureus","Streptococcus pneumoniae","Klebsiella pneumoniae","Streptococcus pneumoniae","Neisseria gonorrhoeae","Pseudomonas aeruginosa","Acinetobacter baumannii","Klebsiella pneumoniae","Klebsiella pneumoniae","Klebsiella pneumoniae","Escherichia coli","Salmonella enterica","Acinetobacter baumannii","Acinetobacter baumannii","Escherichia coli","Neisseria gonorrhoeae","Staphylococcus aureus"],["PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA","PAIRED_ILLUMINA",null,"SINGLE_PACBIO_SMRT",null,"SINGLE_PACBIO_SMRT","SINGLE_PACBIO_SMRT",null,null,"SINGLE_LS454",null,"PAIRED_LS454","SINGLE_ILLUMINA","PAIRED_PACBIO_SMRT","SINGLE_PACBIO_SMRT","SINGLE_PACBIO_SMRT","PAIRED_PACBIO_SMRT","SINGLE_LS454",null,"SINGLE_PACBIO_SMRT","PAIRED_PACBIO_SMRT","SINGLE_OXFORD_NANOPORE","PAIRED_PACBIO_SMRT","SINGLE_PACBIO_SMRT","SINGLE_LS454"],[754,731,636,550,538,532,478,472,458,59,32,27,23,19,18,18,16,15,14,9,9,7,7,6,5,4,4,3,2,2,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>scientific_name_CAMDA<\/th>\n      <th>read_type<\/th>\n      <th>n_accession<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"scientific_name_CAMDA","targets":1},{"name":"read_type","targets":2},{"name":"n_accession","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</section>
<section id="sra-de-las-especies" class="level3" data-number="3.2.3">
<h3 data-number="3.2.3" class="anchored" data-anchor-id="sra-de-las-especies"><span class="header-section-number">3.2.3</span> SRA de las especies</h3>
<p>Cargar funcion</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Separar las especies en variables individuales especie_ids</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"functions/create_species_function.R"</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Funcion que genera archivos txt de solo SRA </span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="st">"functions/Save_ids_files.R"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p>Extraer una variable por especie</p>
<div class="cell">
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Generar una variable por especie</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fu">create_species_variables</span>(SRA_down_joined_data, <span class="st">"scientific_name_CAMDA"</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co"># IDs de las especies descargadas </span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="fu">save_ids</span>(ngonorrhoeae_ids, <span class="st">"ngonorrhoeae"</span>, <span class="at">ruta_salida =</span> <span class="st">"quality_rawdata/species_ids_carpetas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Archivos guardados para ngonorrhoeae</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_ids</span>(abaumannii_ids, <span class="st">"abaumannii"</span>, <span class="at">ruta_salida =</span> <span class="st">"quality_rawdata/species_ids_carpetas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Archivos guardados para abaumannii</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_ids</span>(cjejuni_ids, <span class="st">"cjejuni"</span>, <span class="at">ruta_salida =</span> <span class="st">"quality_rawdata/species_ids_carpetas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Archivos guardados para cjejuni</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_ids</span>(ecoli_ids, <span class="st">"ecoli"</span>, <span class="at">ruta_salida =</span> <span class="st">"quality_rawdata/species_ids_carpetas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Archivos guardados para ecoli</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_ids</span>(kpneumoniae_ids, <span class="st">"kpneumoniae"</span>, <span class="at">ruta_salida =</span> <span class="st">"quality_rawdata/species_ids_carpetas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Archivos guardados para kpneumoniae</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_ids</span>(paeruginosa_ids, <span class="st">"paeruginosa"</span>, <span class="at">ruta_salida =</span> <span class="st">"quality_rawdata/species_ids_carpetas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Archivos guardados para paeruginosa</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_ids</span>(saureus_ids, <span class="st">"saureus"</span>, <span class="at">ruta_salida =</span> <span class="st">"quality_rawdata/species_ids_carpetas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Archivos guardados para saureus</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_ids</span>(senterica_ids, <span class="st">"senterica"</span>, <span class="at">ruta_salida =</span> <span class="st">"quality_rawdata/species_ids_carpetas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Archivos guardados para senterica</code></pre>
</div>
<details open="" class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_ids</span>(spneumoniae_ids, <span class="st">"spneumoniae"</span>, <span class="at">ruta_salida =</span> <span class="st">"quality_rawdata/species_ids_carpetas"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Archivos guardados para spneumoniae</code></pre>
</div>
</div>
<p>Subir los archivos en el cluster</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">scp</span> scripts/move_ids.sh ecoss@elisa.matmor.unam.mx:/files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/scripts</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Script: Crear symlinks de los rawdata (raw/) y almacenarlo por especies en rawspecies/</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a><span class="fu">scp</span> scripts/links_rawdata_byspecies.sh ecoss@elisa.matmor.unam.mx:/files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/scripts</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Script: Crear symlinks de los fastqc (quality/fastqc/) y almacenarlo por especies en quality_byspecies/</span></span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a><span class="fu">scp</span> scripts/links_fastqc_byspecies.sh ecoss@elisa.matmor.unam.mx:/files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/scripts</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Subir la carpeta con los IDs</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">scp</span> <span class="at">-v</span> <span class="at">-r</span> quality_rawdata/species_ids_carpetas ecoss@elisa.matmor.unam.mx:/files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/scripts</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Cargar funcion y crear symlinks de los rawdata</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ecoss@elisa:/files/CAMDA2025_AntibioticResistance/genome_reassembly/reads</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="co"># cargar funcion</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> scripts/move_ids.sh <span class="co"># no tengo permisos</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> scripts/links_rawdata_byspecies.sh</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="co"># ecoss@elisa:/files/CAMDA2025_AntibioticResistance/genome_reassembly/reads</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/abaumannii_test_ids.txt test abaumannii</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/abaumannii_training_ids.txt training abaumannii</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/cjejuni_test_ids.txt test cjejuni</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/cjejuni_training_ids.txt training cjejuni</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/ecoli_test_ids.txt test ecoli</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/ecoli_training_ids.txt training ecoli</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a><span class="co"># link_ids scripts/species_ids_carpetas/kpneumoniae_test_ids.txt test kpneumoniae</span></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/kpneumoniae_training_ids.txt training kpneumoniae</span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/ngonorrhoeae_test_ids.txt test ngonorrhoeae</span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/ngonorrhoeae_training_ids.txt training ngonorrhoeae</span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/paeruginosa_test_ids.txt test paeruginosa</span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/paeruginosa_training_ids.txt training paeruginosa</span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/saureus_test_ids.txt test saureus</span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/saureus_training_ids.txt training saureus</span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/senterica_test_ids.txt test senterica</span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/senterica_training_ids.txt training senterica</span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a><span class="co">#</span></span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/spneumoniae_test_ids.txt test spneumoniae</span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a><span class="ex">link_ids</span> scripts/species_ids_carpetas/spneumoniae_training_ids.txt training spneumoniae</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Informacion de las carpetas contenidas:</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-la</span> raw_byspecies/<span class="pp">*</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="ex">raw_byspecies/abaumannii:</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="ex">total</span> 28</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  4 ecoss bio  4096 jun 25 23:04 .</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span> 11 ecoss bio  4096 jun 25 23:05 ..</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio  4096 jun 25 23:03 test</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 16384 jun 25 23:04 training</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a><span class="ex">raw_byspecies/cjejuni:</span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a><span class="ex">total</span> 52</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  4 ecoss bio  4096 jun 25 23:04 .</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span> 11 ecoss bio  4096 jun 25 23:05 ..</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 16384 jun 25 23:04 test</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 24576 jun 25 23:04 training</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a><span class="ex">raw_byspecies/ecoli:</span></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a><span class="ex">total</span> 48</span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  4 ecoss bio  4096 jun 25 23:05 .</span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span> 11 ecoss bio  4096 jun 25 23:05 ..</span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 20480 jun 25 23:05 test</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 20480 jun 25 23:05 training</span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a><span class="ex">raw_byspecies/kpneumoniae:</span></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a><span class="ex">total</span> 12</span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  3 ecoss bio 4096 jun 25 23:05 .</span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span> 11 ecoss bio 4096 jun 25 23:05 ..</span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 4096 jun 25 23:05 training</span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a><span class="ex">raw_byspecies/ngonorrhoeae:</span></span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a><span class="ex">total</span> 48</span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  4 ecoss bio  4096 jun 25 23:05 .</span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span> 11 ecoss bio  4096 jun 25 23:05 ..</span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio  4096 jun 25 23:05 test</span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 32768 jun 25 23:05 training</span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a><span class="ex">raw_byspecies/paeruginosa:</span></span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a><span class="ex">total</span> 16</span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  4 ecoss bio 4096 jun 25 23:05 .</span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span> 11 ecoss bio 4096 jun 25 23:05 ..</span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 4096 jun 25 23:05 test</span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 4096 jun 25 23:05 training</span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a><span class="ex">raw_byspecies/saureus:</span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a><span class="ex">total</span> 64</span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  4 ecoss bio  4096 jun 25 23:05 .</span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span> 11 ecoss bio  4096 jun 25 23:05 ..</span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 24576 jun 25 23:05 test</span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 24576 jun 25 23:05 training</span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-50"><a href="#cb40-50" aria-hidden="true" tabindex="-1"></a><span class="ex">raw_byspecies/senterica:</span></span>
<span id="cb40-51"><a href="#cb40-51" aria-hidden="true" tabindex="-1"></a><span class="ex">total</span> 16</span>
<span id="cb40-52"><a href="#cb40-52" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  4 ecoss bio 4096 jun 25 23:05 .</span>
<span id="cb40-53"><a href="#cb40-53" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span> 11 ecoss bio 4096 jun 25 23:05 ..</span>
<span id="cb40-54"><a href="#cb40-54" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 4096 jun 25 23:05 test</span>
<span id="cb40-55"><a href="#cb40-55" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 4096 jun 25 23:05 training</span>
<span id="cb40-56"><a href="#cb40-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-57"><a href="#cb40-57" aria-hidden="true" tabindex="-1"></a><span class="ex">raw_byspecies/spneumoniae:</span></span>
<span id="cb40-58"><a href="#cb40-58" aria-hidden="true" tabindex="-1"></a><span class="ex">total</span> 40</span>
<span id="cb40-59"><a href="#cb40-59" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  4 ecoss bio  4096 jun 25 23:05 .</span>
<span id="cb40-60"><a href="#cb40-60" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span> 11 ecoss bio  4096 jun 25 23:05 ..</span>
<span id="cb40-61"><a href="#cb40-61" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 16384 jun 25 23:05 test</span>
<span id="cb40-62"><a href="#cb40-62" aria-hidden="true" tabindex="-1"></a><span class="ex">drwxr-xr-x</span>  2 ecoss bio 16384 jun 25 23:05 training</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Cargar funcion y crear symlink de los fastqc por especies</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># ecoss@elisa:/files/CAMDA2025_AntibioticResistance/genome_reassembly/reads</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="co"># cargar funcion</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="bu">source</span> scripts/links_fastqc_byspecies.sh</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; abaumannii</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/abaumannii_test_ids.txt test abaumannii <span class="co"># sin reportes</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/abaumannii_training_ids.txt training abaumannii</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a><span class="co"># correr multiqc</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> quality_byspecies/abaumannii/training <span class="at">-n</span> multiqc_abaumannii_training.html <span class="at">-o</span> quality_byspecies/multiqc_report</span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a><span class="co"># /// MultiQC 🔍 v1.29</span></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a><span class="co">#        file_search | Search path: /files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/quality_byspecies/abaumannii/training</span></span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a><span class="co">#          searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 1264/1264  </span></span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a><span class="co">#             fastqc | Found 632 reports</span></span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a><span class="co">#      write_results | Data        : quality_byspecies/multiqc_report/multiqc_abaumannii_training_data</span></span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a><span class="co">#      write_results | Report      : quality_byspecies/multiqc_report/multiqc_abaumannii_training.html</span></span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a><span class="co">#            multiqc | MultiQC complete</span></span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; cjejuni</span></span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/cjejuni_test_ids.txt test cjejuni <span class="co"># sin reportes link_fastqc_by_species scripts/species_ids_carpetas/cjejuni_training_ids.txt training cjejuni</span></span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a><span class="co"># correr multiqc</span></span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> quality_byspecies/cjejuni/training <span class="at">-n</span> multiqc_cjejuni_training.html <span class="at">-o</span> quality_byspecies/multiqc_report</span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a><span class="co"># /// MultiQC 🔍 v1.29</span></span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a><span class="co">#        file_search | Search path: /files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/quality_byspecies/cjejuni/training</span></span>
<span id="cb41-30"><a href="#cb41-30" aria-hidden="true" tabindex="-1"></a><span class="co">#          searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 2132/2132  </span></span>
<span id="cb41-31"><a href="#cb41-31" aria-hidden="true" tabindex="-1"></a><span class="co">#             fastqc | Found 1066 reports</span></span>
<span id="cb41-32"><a href="#cb41-32" aria-hidden="true" tabindex="-1"></a><span class="co">#      write_results | Data        : quality_byspecies/multiqc_report/multiqc_cjejuni_training_data</span></span>
<span id="cb41-33"><a href="#cb41-33" aria-hidden="true" tabindex="-1"></a><span class="co">#      write_results | Report      : quality_byspecies/multiqc_report/multiqc_cjejuni_training.html</span></span>
<span id="cb41-34"><a href="#cb41-34" aria-hidden="true" tabindex="-1"></a><span class="co">#            multiqc | MultiQC complete</span></span>
<span id="cb41-35"><a href="#cb41-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-36"><a href="#cb41-36" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; ecoli</span></span>
<span id="cb41-37"><a href="#cb41-37" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/ecoli_test_ids.txt test ecoli</span>
<span id="cb41-38"><a href="#cb41-38" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/ecoli_training_ids.txt training ecoli</span>
<span id="cb41-39"><a href="#cb41-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-40"><a href="#cb41-40" aria-hidden="true" tabindex="-1"></a><span class="co"># correr multiqc</span></span>
<span id="cb41-41"><a href="#cb41-41" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> quality_byspecies/ecoli/training <span class="at">-n</span> multiqc_ecoli_training.html <span class="at">-o</span> quality_byspecies/multiqc_report</span>
<span id="cb41-42"><a href="#cb41-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-43"><a href="#cb41-43" aria-hidden="true" tabindex="-1"></a><span class="co"># /// MultiQC 🔍 v1.29</span></span>
<span id="cb41-44"><a href="#cb41-44" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb41-45"><a href="#cb41-45" aria-hidden="true" tabindex="-1"></a><span class="co">#        file_search | Search path: /files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/quality_byspecies/ecoli/training</span></span>
<span id="cb41-46"><a href="#cb41-46" aria-hidden="true" tabindex="-1"></a><span class="co">#          searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 1894/1894  </span></span>
<span id="cb41-47"><a href="#cb41-47" aria-hidden="true" tabindex="-1"></a><span class="co">#             fastqc | Found 947 reports</span></span>
<span id="cb41-48"><a href="#cb41-48" aria-hidden="true" tabindex="-1"></a><span class="co">#      write_results | Data        : quality_byspecies/multiqc_report/multiqc_ecoli_training_data</span></span>
<span id="cb41-49"><a href="#cb41-49" aria-hidden="true" tabindex="-1"></a><span class="co">#      write_results | Report      : quality_byspecies/multiqc_report/multiqc_ecoli_training.html</span></span>
<span id="cb41-50"><a href="#cb41-50" aria-hidden="true" tabindex="-1"></a><span class="co">#            multiqc | MultiQC complete</span></span>
<span id="cb41-51"><a href="#cb41-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-52"><a href="#cb41-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-53"><a href="#cb41-53" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; kpneumoniae - sin reporte en ambos</span></span>
<span id="cb41-54"><a href="#cb41-54" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/kpneumoniae_test_ids.txt test kpneumoniae</span>
<span id="cb41-55"><a href="#cb41-55" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/kpneumoniae_training_ids.txt training kpneumoniae</span>
<span id="cb41-56"><a href="#cb41-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-57"><a href="#cb41-57" aria-hidden="true" tabindex="-1"></a><span class="co"># correr multiqc</span></span>
<span id="cb41-58"><a href="#cb41-58" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> quality_byspecies/kpneumoniae/training <span class="at">-n</span> multiqc_kpneumoniae_training.html <span class="at">-o</span> quality_byspecies/multiqc_report</span>
<span id="cb41-59"><a href="#cb41-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-60"><a href="#cb41-60" aria-hidden="true" tabindex="-1"></a>           </span>
<span id="cb41-61"><a href="#cb41-61" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; ngonorrhoeae</span></span>
<span id="cb41-62"><a href="#cb41-62" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/ngonorrhoeae_test_ids.txt test ngonorrhoeae</span>
<span id="cb41-63"><a href="#cb41-63" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/ngonorrhoeae_training_ids.txt training ngonorrhoeae</span>
<span id="cb41-64"><a href="#cb41-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-65"><a href="#cb41-65" aria-hidden="true" tabindex="-1"></a><span class="co"># correr multiqc</span></span>
<span id="cb41-66"><a href="#cb41-66" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> quality_byspecies/ngonorrhoeae/training <span class="at">-n</span> multiqc_ngonorrhoeae_training.html <span class="at">-o</span> quality_byspecies/multiqc_report</span>
<span id="cb41-67"><a href="#cb41-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-68"><a href="#cb41-68" aria-hidden="true" tabindex="-1"></a><span class="co"># /// MultiQC 🔍 v1.29</span></span>
<span id="cb41-69"><a href="#cb41-69" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb41-70"><a href="#cb41-70" aria-hidden="true" tabindex="-1"></a><span class="co">#        file_search | Search path: /files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/quality_byspecies/ngonorrhoeae/training</span></span>
<span id="cb41-71"><a href="#cb41-71" aria-hidden="true" tabindex="-1"></a><span class="co">#          searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 2836/2836  </span></span>
<span id="cb41-72"><a href="#cb41-72" aria-hidden="true" tabindex="-1"></a><span class="co">#             fastqc | Found 1418 reports</span></span>
<span id="cb41-73"><a href="#cb41-73" aria-hidden="true" tabindex="-1"></a><span class="co">#      write_results | Data        : quality_byspecies/multiqc_report/multiqc_ngonorrhoeae_training_data</span></span>
<span id="cb41-74"><a href="#cb41-74" aria-hidden="true" tabindex="-1"></a><span class="co">#      write_results | Report      : quality_byspecies/multiqc_report/multiqc_ngonorrhoeae_training.html</span></span>
<span id="cb41-75"><a href="#cb41-75" aria-hidden="true" tabindex="-1"></a><span class="co">#            multiqc | MultiQC complete </span></span>
<span id="cb41-76"><a href="#cb41-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-77"><a href="#cb41-77" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; Sin reporte de fastqc</span></span>
<span id="cb41-78"><a href="#cb41-78" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/paeruginosa_test_ids.txt test paeruginosa</span>
<span id="cb41-79"><a href="#cb41-79" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/paeruginosa_training_ids.txt training paeruginosa</span>
<span id="cb41-80"><a href="#cb41-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-81"><a href="#cb41-81" aria-hidden="true" tabindex="-1"></a><span class="co"># correr multiqc</span></span>
<span id="cb41-82"><a href="#cb41-82" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> quality_byspecies/paeruginosa/training <span class="at">-n</span> multiqc_paeruginosa_training.html <span class="at">-o</span> quality_byspecies/multiqc_report</span>
<span id="cb41-83"><a href="#cb41-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-84"><a href="#cb41-84" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; saureus</span></span>
<span id="cb41-85"><a href="#cb41-85" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/saureus_test_ids.txt test saureus</span>
<span id="cb41-86"><a href="#cb41-86" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/saureus_training_ids.txt training saureus</span>
<span id="cb41-87"><a href="#cb41-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-88"><a href="#cb41-88" aria-hidden="true" tabindex="-1"></a><span class="co"># correr multiqc</span></span>
<span id="cb41-89"><a href="#cb41-89" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> quality_byspecies/saureus/training <span class="at">-n</span> multiqc_saureus_training.html <span class="at">-o</span> quality_byspecies/multiqc_report</span>
<span id="cb41-90"><a href="#cb41-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-91"><a href="#cb41-91" aria-hidden="true" tabindex="-1"></a><span class="co"># /// MultiQC 🔍 v1.29</span></span>
<span id="cb41-92"><a href="#cb41-92" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb41-93"><a href="#cb41-93" aria-hidden="true" tabindex="-1"></a><span class="co">#        file_search | Search path: /files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/quality_byspecies/saureus/training</span></span>
<span id="cb41-94"><a href="#cb41-94" aria-hidden="true" tabindex="-1"></a><span class="co">#          searching | ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 1796/1796  </span></span>
<span id="cb41-95"><a href="#cb41-95" aria-hidden="true" tabindex="-1"></a><span class="co">#             fastqc | Found 898 reports</span></span>
<span id="cb41-96"><a href="#cb41-96" aria-hidden="true" tabindex="-1"></a><span class="co">#      write_results | Data        : quality_byspecies/multiqc_report/multiqc_saureus_training_data</span></span>
<span id="cb41-97"><a href="#cb41-97" aria-hidden="true" tabindex="-1"></a><span class="co">#      write_results | Report      : quality_byspecies/multiqc_report/multiqc_saureus_training.html</span></span>
<span id="cb41-98"><a href="#cb41-98" aria-hidden="true" tabindex="-1"></a><span class="co">#            multiqc | MultiQC complete</span></span>
<span id="cb41-99"><a href="#cb41-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-100"><a href="#cb41-100" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; senterica</span></span>
<span id="cb41-101"><a href="#cb41-101" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/senterica_test_ids.txt test senterica</span>
<span id="cb41-102"><a href="#cb41-102" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/senterica_training_ids.txt training senterica</span>
<span id="cb41-103"><a href="#cb41-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-104"><a href="#cb41-104" aria-hidden="true" tabindex="-1"></a><span class="co"># correr multiqc</span></span>
<span id="cb41-105"><a href="#cb41-105" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> quality_byspecies/senterica/training <span class="at">-n</span> multiqc_senterica_training.html <span class="at">-o</span> quality_byspecies/multiqc_report</span>
<span id="cb41-106"><a href="#cb41-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-107"><a href="#cb41-107" aria-hidden="true" tabindex="-1"></a><span class="co"># &gt; spneumoniae</span></span>
<span id="cb41-108"><a href="#cb41-108" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/spneumoniae_test_ids.txt test spneumoniae</span>
<span id="cb41-109"><a href="#cb41-109" aria-hidden="true" tabindex="-1"></a><span class="ex">link_fastqc_by_species</span> scripts/species_ids_carpetas/spneumoniae_training_ids.txt training spneumoniae</span>
<span id="cb41-110"><a href="#cb41-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-111"><a href="#cb41-111" aria-hidden="true" tabindex="-1"></a><span class="co"># correr multiqc</span></span>
<span id="cb41-112"><a href="#cb41-112" aria-hidden="true" tabindex="-1"></a><span class="ex">multiqc</span> quality_byspecies/spneumoniae/training <span class="at">-n</span> multiqc_spneumoniae_training.html <span class="at">-o</span> quality_byspecies/multiqc_report</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Descargar archivos</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># /Users/ecoss/Documents/CAMDA2025_metadatos/quality_rawdata</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">scp</span> <span class="at">-r</span> ecoss@elisa.matmor.unam.mx:/files/CAMDA2025_AntibioticResistance/genome_reassembly/reads/quality_byspecies/multiqc_report/<span class="pp">*</span>html .</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<blockquote class="blockquote">
<p>one of the Staphylococcus aureus accession codes is incorrect, as follows: <strong><em>SRR2101499 should be ERR1197981</em></strong> <a href="https://www.nature.com/articles/ncomms11465">Bradley, et al.&nbsp;2016. Nature</a></p>
<p>De acuerdo a <a href="https://www.ebi.ac.uk/ena/browser/text-search?query=SRR2101499">ENA</a>, este <strong><em>SRR2101499</em></strong> corresponde a <em>Mycobacterium tuberculosis</em>.</p>
</blockquote>
<p>SRR1187837 y SRR1187838 - E.coli UCI 54</p>
<p>SRR3982215 - E.coli</p>
<p>SRR3996243</p>
</section>
</section>
<section id="eliminacion-de-secuencias-de-mala-calidad-y-de-adaptadores" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="eliminacion-de-secuencias-de-mala-calidad-y-de-adaptadores"><span class="header-section-number">3.3</span> Eliminacion de secuencias de mala calidad y de adaptadores</h2>
<p>Trim Galore! o Trimmomatic - Solo illumina</p>
<p>Filtlong o NanoFilt - Oxford Nanopore / PacBio</p>
<p>Cutadapt o Prinseq-lite - SOLiD / Ion Torrent / 454</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./2_Preprocessing_CAMDApublicdata.html" class="pagination-link" aria-label="Comparing CAMDA metadata related with public data">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Comparing CAMDA metadata related with public data</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./4_inpufile_MLandDL.html" class="pagination-link" aria-label="Generating the Input File for Machine Learning and Deep Learning Models">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Generating the Input File for Machine Learning and Deep Learning Models</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>